core/services/relay/evm/mercury/wsrpc/client.go

```diff
@@ -161,8 +189,10 @@ func (w *client) dial(ctx context.Context, opts ...wsrpc.DialOption) error {
 	}
 	w.dialSuccessCountMetric.Inc()
 	setLivenessMetric(true)
+	w.mu.Lock()
 	w.conn = conn
 	w.rawClient = pb.NewMercuryClient(conn)
+	w.mu.Unlock()
```

Comment:
can `NewMercuryClient` throw any errors? Do you just want to move this above and `defer` it?